tests/components/here_weather/test_init.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for the here_weather integration."""
+from datetime import timedelta
+from unittest.mock import patch
+
+from homeassistant.components.here_weather.const import DEFAULT_SCAN_INTERVAL, DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+import homeassistant.util.dt as dt_util
+
+from . import mock_weather_for_coordinates
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_number_of_listeners(hass):
+    """Test that number_of_listerners works."""
+    utcnow = dt_util.utcnow()
+    # Patching 'utcnow' to gain more control over the timed update.
+    with patch("homeassistant.util.dt.utcnow", return_value=utcnow):
+        with patch(
+            "herepy.DestinationWeatherApi.weather_for_coordinates",
+            side_effect=mock_weather_for_coordinates,
+        ):
+            entry = MockConfigEntry(
+                domain=DOMAIN,
+                data={
+                    CONF_API_KEY: "test",
+                    CONF_NAME: DOMAIN,
+                    CONF_LATITUDE: "40.79962",
+                    CONF_LONGITUDE: "-73.970314",
+                },
+            )
+            entry.add_to_hass(hass)
+
+            await hass.config_entries.async_setup(entry.entry_id)
+
+            await hass.async_block_till_done()
+
+            sensor = hass.states.get("weather.here_weather_forecast_7days_simple")
+            assert sensor.state == "cloudy"
+            async_fire_time_changed(hass, utcnow + timedelta(DEFAULT_SCAN_INTERVAL * 2))
+            await hass.async_block_till_done()
+            sensor = hass.states.get("weather.here_weather_forecast_7days_simple")
+            assert sensor.state == "cloudy"
+
+
+async def test_update_interval(hass):
+    """Test that update_interval is correctly set."""
+    utcnow = dt_util.utcnow()
+    # Patching 'utcnow' to gain more control over the timed update.
+    with patch("homeassistant.util.dt.utcnow", return_value=utcnow):
+        with patch(
+            "herepy.DestinationWeatherApi.weather_for_coordinates",
+            side_effect=mock_weather_for_coordinates,
+        ):
+            entry = MockConfigEntry(
+                domain=DOMAIN,
+                data={
+                    CONF_API_KEY: "test",
+                    CONF_NAME: DOMAIN,
+                    CONF_LATITUDE: "40.79962",
+                    CONF_LONGITUDE: "-73.970314",
+                },
+            )
+            entry.add_to_hass(hass)
+
+            await hass.config_entries.async_setup(entry.entry_id)
+
+            await hass.async_block_till_done()
+
+            sensor = hass.states.get("weather.here_weather_forecast_7days_simple")
+            assert sensor.state == "cloudy"
+            with patch(
+                "homeassistant.components.here_weather.active_here_clients",
```

Comment:
It's better to set up multiple config entries to make sure all our logic really works. Also please test some different cases with different results.

Reply:
I can implement a utility test method to set up 200 integrations instead.
What different results would you test for in this specific test?

Reply:
Adding a test of the util is ok but I'd also like to see more test cases here where we test that the update interval changes when we add another config entry. I'd test at least 0, 1 and more than 1 config entry.

Reply:
The default update rate is every 300s. It will only be slower when more than 27 integrations with all entities enabled are registered.

Reply:
We can also just keep the default rate and remove the throttling. I would need to adjust it anyway as soon as the `here_travel_time` config_flow PR gets merged.

Reply:
Yes, if we don't believe that it's realistically worth the effort for the extra calculation we can remove it.

Reply:
Removed in https://github.com/home-assistant/core/pull/28910/commits/1720b37b715d79fbb70b4a4ae9c033a14581c7c4